File: zelda/src/main/java/com/pel/zelda/model/Team.java
package com.pel.zelda.model;

import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * User: bharat
 * Date: 8/10/21
 * Time: 1:23 PM
 */
public class Team {

    public Integer id;
    public String name;
    public String logoUrl;
    public String game;
    public String avgRank;
    public Timestamp createdAt;
    public Timestamp updatedAt;

    public List<Map<String, Object>> users = new ArrayList<>();
}

File: lib/models/team.dart
class Team {
  int? id;
  String? name;
  String? logoUrl;
  String? game;
  String? avgRank;

  DateTime? createdAt;
  DateTime? updatedAt;

  Team();

  Team.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    logoUrl = json['logoUrl'];
    game = json['game'];
    avgRank = json['avgRank'];
    createdAt = DateTime.tryParse(json['createdAt']);
    updatedAt = DateTime.tryParse(json['updatedAt']);
  }

  Map<String, dynamic> toJson() => {
    'id': id ?? "null",
    'name': name ?? "null",
    'logoUrl': logoUrl ?? "null",
    'game': game ?? "null",
    'avgRank': avgRank ?? "null",
    'createdAt': createdAt != null ? createdAt.toString() : "null",
    'updatedAt': updatedAt != null ? updatedAt.toString() : "null",
  };
}